transport-native-epoll/src/main/c/netty_unix_socket.c

```diff
@@ -788,6 +811,26 @@ jint netty_unix_socket_JNI_OnLoad(JNIEnv* env, const char* packagePrefix) {
         netty_unix_errors_throwRuntimeException(env, "failed to get method ID: NetUild.isIpV4StackPreferred()");
         return JNI_ERR;
     }
+    
+    nettyClassName = netty_unix_util_prepend(packagePrefix, "io/netty/channel/unix/PeerCredentials");
+    jclass localPeerCredsClass = (*env)->FindClass(env, nettyClassName);
+    free(nettyClassName);
+    nettyClassName = NULL;
+    if (localPeerCredsClass == NULL) {
+        // pending exception...
+        return JNI_ERR;
+    }
+    peerCredentialsClass = (jclass) (*env)->NewGlobalRef(env, localPeerCredsClass);
```

Comment:
@cmdc0de you create a global reference here. This also needs to be destroyed at some point to not leak native memory.